core/src/main/java/com/linecorp/armeria/internal/common/util/StringUtil.java

```diff
@@ -68,7 +67,11 @@ public static Boolean toBoolean(String s) {
 
     @Nullable
     public static Boolean toBooleanOrNull(String s) {
-        return stringToBoolean.get(s);
+        if (s == null) {
+            return null;
+        }
```

Comment:
As all packages are annotated with `@NonNullByDefault`, I think it's safe to assume that the input string will be non-null.

```suggestion
```